lib/spack/spack/binary_distribution.py

```diff
@@ -256,7 +256,7 @@ def build_tarball(spec, outdir, force=False, rel=False, yes_to_all=False,
     # create compressed tarball of the install prefix
     with closing(tarfile.open(tarfile_path, 'w:gz')) as tar:
         tar.add(name='%s' % workdir,
-                arcname='%s' % os.path.basename(workdir))
```

Comment:
This appears to work but isn't clear from https://docs.python.org/2/library/tarfile.html - could you add a comment mentioning that this creates an exploding tarball with all the files in `workdir` but not `workdir` itself? IMO it's also worthwhile to mention that you do this to support installing binary packages on a spack instance with a different directory layout.